### PR TITLE
Add funding report table

### PR DESCRIPTION
**Pull request by nickbristow**
## Background
Adds new model and table for funding report.

ID (uuid, unique)
Month (date)
Funding Source (enum)
Site (foreign key on Site table)
Organization (foreign key on Organization table)
Report (json)

## GitHub Issue
- #1430 

## Validation Plan
- [x] Run migration, check that funding report table is created, make sure all columns are created

## Automated Testing
- [x] All unit tests are passing.

### CODE REVIEW

File: src/entity/FundingReport.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  report: String;
+
+  @OneToOne(() => Organization, { nullable: false })
+  organization: Organization;
+
+  @OneToOne(() => Site, { nullable: false })
+  site: Site;
+
+  @Column({ type: 'date', transformer: momentTransformer })
+  month: Moment;
```
[the remark this answers is not in the document]
Comment (JordanGuinn) [reply src/entity/FundingReport.ts:42]: I actually think `month` probably makes the most sense as the column name here - even though it's still being stored as `datetime2`, it's more indicative of what the intent of the field is.  If we called this `date`, that could easily be interpreted as the time this report was generated, which is wrong

File: src/entity/FundingReport.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  @Column()
+  enrollmentId: number;
```
[the remark this answers is not in the document]
Comment (nickbristow) [reply src/entity/FundingReport.ts]: copy pasted by mistake, removed

File: src/entity/FundingReport.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export class FundingReport implements FundingReportInterface {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column({
+    type: 'varchar',
+    length: 20,
+    transformer: enumTransformer(FundingSource),
+  })
+  fundingSource: FundingSource;
```
[the remark this answers is not in the document]
Comment (nickbristow) [reply src/entity/FundingReport.ts:31]: added, not nullable

**Review by bamader — APPROVED** [review]
Changes look good to me here. I'm happy to move forward with validating this!

**Comment by bamader** [comment]
## Validation Plan

- Build app with freshly blasted `dist`
- Log in
- [ ] Check that a `FundingReport` table has been created with all fields specified in the entity schema

File: src/entity/FundingReport.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Column({
+    type: 'varchar',
+    length: 20,
+    transformer: enumTransformer(FundingSource),
+    nullable: false,
+  })
+  fundingSource: FundingSource;
+
+  @Column({ type: 'text' })
```
Comment (JordanGuinn) [inline src/entity/FundingReport.ts:33]: nit: Given that mssql doesn't actually have direct support for JSON columns, can we actually use the [simpleJSON column type](text) here instead?

Comment (bamader) [reply src/entity/FundingReport.ts:33]: Wait waaaaaaat? I swear it had JSON support?

Comment (bamader) [reply src/entity/FundingReport.ts:33]: Huh. I guess not. Nvm.

Comment (JordanGuinn) [reply src/entity/FundingReport.ts:33]: Ehhhh, it kinda does - it looks like you can use [built-ins to transform your text data into JSON on the fly](https://docs.microsoft.com/en-us/sql/relational-databases/json/json-data-sql-server?view=sql-server-ver15), but unlike MySQL and Postgres, it doesn't appear that you can actually _store_ data as JSON 😕

Comment (JordanGuinn) [reply src/entity/FundingReport.ts:33]: Gotta love SQL Server 😭

Comment (bamader) [reply src/entity/FundingReport.ts:33]: @nickbristow If you're cool with it, I can just make this change myself and then continue with validation?

**Comment by bamader** [comment]
This looks good to me! Had some troubles with my own local set up actually getting the migrations to apply to the DB, but not the fault of this PR at all. Ready to merge!

**Review by lukenewman — APPROVED** [review]

**Pull request merged**